src/aliceVision/multiview/essentialF10Solver.hpp

```diff
@@ -0,0 +1,235 @@
+// This file is part of the AliceVision project.
+// Copyright (c) 2018 AliceVision contributors.
+// Copyright (c) 2015 Jan Heller <hellej1@cmp.felk.cvut.cz>, Zuzana Kukelova <zukuke@microsoft.com>
+// This Source Code Form is subject to the terms of the Mozilla Public License,
+// v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <aliceVision/numeric/numeric.hpp>
+
+#include <Eigen/Dense>
+
+#include <cmath>
+#include <limits>
+#include <vector>
+
+namespace aliceVision {
+
+using Mat21 = Eigen::Matrix<double, 2, 1>;
+
+template <typename Scalar>
+bool isZero(Scalar val)
+{
+  return (+val < (100.0 * std::numeric_limits<Scalar>::epsilon())) &&
+         (-val < (100.0 * std::numeric_limits<Scalar>::epsilon()));
+}
+
+/**
+ * @brief	Computes column echelon form of a matrix
+ * @author Jan Heller, adapted to aliceVision by Michal Polic
+ */
+template <typename Derived>
+inline void colEchelonForm(Eigen::MatrixBase<Derived>& M, double pivtol = 1e-12)
+{
+  using Scalar = typename Derived::Scalar;
+
+  const int n = M.rows();
+  const int m = M.cols();
+
+  int i = 0;
+  int j = 0;
+  int k = 0;
+  int col = 0;
+
+  Scalar p, tp;
+
+  while((i < m) && (j < n))
+  {
+    p = std::numeric_limits<Scalar>::min();
+    col = i;
+
+    for(k = i; k < m; k++)
+    {
+      tp = std::abs(M(j, k));
+      if(tp > p)
+      {
+        p = tp;
+        col = k;
+      }
+    }
+
+    if(p < Scalar(pivtol))
+    {
+      M.block(j, i, 1, m - i).setZero();
+      ++j;
+    }
+    else
+    {
+      if(col != i)
+        M.block(j, i, n - j, 1).swap(M.block(j, col, n - j, 1));
+
+      M.block(j + 1, i, n - j - 1, 1) /= M(j, i);
+      M(j, i) = 1.0;
+
+      for(k = 0; k < m; k++)
+      {
+        if(k == i)
+          continue;
+
+        M.block(j, k, n - j, 1) -= M(j, k) * M.block(j, i, n - j, 1);
+      }
+
+      ++i;
+      ++j;
+    }
+  }
+}
+
+template <typename Scalar, typename Derived>
+int f10e_gb(Eigen::Matrix<Scalar, 29, 1>& pr, Eigen::MatrixBase<Derived>& sols, double pivtol = 1e-16)
```

Comment:
`pr` seems to be `const` 
Also a more meaningful name... Maybe `params` as it is used in the call?